internal/literals/literals.go

```diff
@@ -129,9 +156,19 @@ func Obfuscate(files []*ast.File, info *types.Info, fset *token.FileSet, blackli
 			}
 
 			switch x.Kind {
-			case token.FLOAT, token.INT:
-				obfuscateNumberLiteral(cursor, info)
+			case token.INT:
```

Comment:
Note this is not the actual type, a float value can be a token.Int and vice versa.

`var x int = 1.0` is token.Float


`var x float64 = 1` is token.Int

You need to move this into the obfuscateNumberLiteral func.

Reply:
Ok, so I think I need to modify `obfuscateNumberLiteral` to return `(bool, error)` so that I can return `false` if an int,float shouldn't be obfuscated for the `astutil.Apply` post function. Also, I see that you don't handle errors from `obfuscateNumberLiteral`, so for now I'll just panic like you do already if `strconv.Unquote` fails

Reply:
Hmm... by handling errors from `obfuscateNumberLiteral`, the `literals.txt` reliably fails with this error: 
```
panic: obfuscating numerical literal failed: strconv.ParseInt: parsing "-122.0"
```

So I think some numeric obfuscation logic may need to be fixed